data-prepper-plugins/saas-source-plugins/microsoft-office365-source/src/test/java/org/opensearch/dataprepper/plugins/source/microsoft_office365/auth/Office365AuthenticationProviderTest.java

```diff
@@ -38,17 +52,26 @@ class Office365AuthenticationProviderTest {
     private AuthenticationConfiguration authConfig;
 
     @Mock
-    private AuthenticationConfiguration.OAuth2Credentials oAuth2Credentials;
+    private Oauth2Config oAuth2Config;
+
+    @Mock
+    private PluginConfigVariable clientIdVariable;
+
+    @Mock
+    private PluginConfigVariable clientSecretVariable;
 
     private Office365AuthenticationProvider authProvider;
 
     @BeforeEach
     void setUp() throws NoSuchFieldException, IllegalAccessException {
-        when(config.getAuthenticationConfiguration()).thenReturn(authConfig);
-        when(authConfig.getOauth2()).thenReturn(oAuth2Credentials);
-        when(oAuth2Credentials.getClientId()).thenReturn("testClientId");
-        when(oAuth2Credentials.getClientSecret()).thenReturn("testClientSecret");
-        when(config.getTenantId()).thenReturn("testTenantId");
+        // Use lenient mocks to avoid UnnecessaryStubbingException
+        lenient().when(config.getAuthenticationConfiguration()).thenReturn(authConfig);
```

Comment:
Avoid using lenient. You should setup the mocks where they are needed to keep the tests clean. You can use `@Nested` test suites to help organize this some.